### PR TITLE
feat: map feature click events

### LINKED ISSUE #2513: [map] Feature click event
### Motivation

The map should provide a way to listen and react to clicking on features in form of a DOM event.

### Acceptance criteria
- Map allows registering event listeners for click events on features
- Click event contains the feature that was clicked on
- Click event contains information about the clicked coordinates
- Click event contains information about the source click event

### PR DESCRIPTION
**Pull request by sissbruecker**
## Description

Allows to get notified about clicks on features. The click events contain the clicked feature, as well as the layer and the source that contain the feature. Optionally a filter layer can be specified for the click listener, so that the listener is only triggered when features in a specific layer are clicked.

Also changes the OL event used for map clicks and feature clicks to `singleclick` in order to ignore double-clicks and drags.

Fixes #2513

### CODE REVIEW

File: vaadin-map-flow-parent/vaadin-map-flow-integration-tests/src/main/java/com/vaadin/flow/component/map/MapView.java [hunk 1]
```diff
@@ -28,6 +28,12 @@ public MapView() {
                 new Coordinate(1233058.1696443919, 6351912.406929109));
         map.getFeatureLayer().addFeature(nurembergMarker);
 
+        map.addFeatureClickListener(e -> {
```
Comment (hdamr) [inline vaadin-map-flow-parent/vaadin-map-flow-integration-tests/src/main/java/com/vaadin/flow/component/map/MapView.java:31]: Is it necessary to have this in this page?

File: vaadin-map-flow-parent/vaadin-map-flow/src/main/resources/META-INF/resources/frontend/vaadin-map/util.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import VectorSource from "ol/source/Vector";
+
+/**
+ * Searches an OpenLayers map instance for the layer whose source contains a specific feature
+ * @param map an OpenLayers map instance
+ * @param feature the feature that should be contained in the layers source
+ * @returns {*} the layer that contains the feature, or null
```
Comment (hdamr) [inline vaadin-map-flow-parent/vaadin-map-flow/src/main/resources/META-INF/resources/frontend/vaadin-map/util.js:7]: `find` returns `undefined` when no element is satisfied with condition.

File: vaadin-map-flow-parent/vaadin-map-flow-integration-tests/src/main/java/com/vaadin/flow/component/map/MapView.java (hunk 1 above)
Comment (sissbruecker) [reply vaadin-map-flow-parent/vaadin-map-flow-integration-tests/src/main/java/com/vaadin/flow/component/map/MapView.java:31]: For me that page was just some sort of playground. Actually I think we can now remove it, or at least replace it with a couple of demos. Would suggest to ignore it for now and then replace it with demos in a separate PR.

File: vaadin-map-flow-parent/vaadin-map-flow/src/main/resources/META-INF/resources/frontend/vaadin-map/util.js (hunk 2 above)
Comment (sissbruecker) [reply vaadin-map-flow-parent/vaadin-map-flow/src/main/resources/META-INF/resources/frontend/vaadin-map/util.js]: Fixed the JSDoc

**Review by hdamr — APPROVED** [review]

**Pull request merged**